src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.StackRowStack.cs

```diff
@@ -67,6 +67,8 @@ private void Enlarge()
                     _rentedBuffer.Length - toReturn.Length + _topOfStack,
                     toReturn.Length - _topOfStack);
 
+                _topOfStack += _rentedBuffer.Length - toReturn.Length;
```

Comment:
@bartonjs, there was a test gap around this `Enlarge()` method. The new tests in this PR exercise that branch now, but we probably want more test cases here. I want to highlight this change explicitly for your review.

![image](https://user-images.githubusercontent.com/6527137/53666635-7e2e5b80-3c23-11e9-8ace-72f072c4f725.png)